Foundation/FileManager.swift

```diff
@@ -992,15 +1005,31 @@ open class FileManager : NSObject {
      */
     open func fileSystemRepresentation(withPath path: String) -> UnsafePointer<Int8> {
         precondition(path != "", "Empty path argument")
+#if os(Windows)
+        return try! _fileSystemRepresentation(withPath: path) {
+            String(decodingCString: $0, as: UTF16.self).withCString() {
+                let sz = strnlen($0, Int(MAX_PATH))
+                let buf = UnsafeMutablePointer<Int8>.allocate(capacity: sz + 1)
+                buf.initialize(from: $0, count: sz + 1)
+                return UnsafePointer(buf)
+            }
+        }
+#else
         return try! __fileSystemRepresentation(withPath: path)
+#endif
     }
 
-    internal func __fileSystemRepresentation(withPath path: String) throws -> UnsafePointer<Int8> {
+    internal func __fileSystemRepresentation(withPath path: String) throws -> UnsafePointer<NativeFSRCharType> {
         let len = CFStringGetMaximumSizeOfFileSystemRepresentation(path._cfObject)
         if len != kCFNotFound {
-            let buf = UnsafeMutablePointer<Int8>.allocate(capacity: len)
+            let buf = UnsafeMutablePointer<NativeFSRCharType>.allocate(capacity: len)
             buf.initialize(repeating: 0, count: len)
-            if path._nsObject.getFileSystemRepresentation(buf, maxLength: len) {
+#if os(Windows)
+            let result = path._nsObject._getFileSystemRepresentation(buf, maxLength: len)
+#else
+            let result = path._nsObject.getFileSystemRepresentation(buf, maxLength: len)
+#endif
```

Comment:
This is not a good way to this as every call site would need to be updated. I would add a wrapper `_getFileSystemRepresentation` for the non-Windows version and just call `_getFileSystemRepresentation` for all  uses.